pkgs/by-name/dn/dnspy/wrapper.nix

```diff
@@ -0,0 +1,86 @@
+{
+  lib,
+  buildPackages,
+  stdenvNoCC,
+  wineWow64Packages,
+  writeShellScript,
+  makeWrapper,
+  copyDesktopItems,
+  makeDesktopItem,
+  nix-update,
+
+  dnspy,
+  is32Bit ? false,
+  overrideWine ? wineWow64Packages.staging,
+}:
+let
+  wine = overrideWine;
+
+  wrapper = writeShellScript "dnspy-wrapper" ''
+    export WINE="${lib.getExe wine}"
+    export WINEPREFIX="''${DNSPY_HOME:-"''${XDG_DATA_HOME:-"''${HOME}/.local/share"}/dnSpy"}/wine"
+    export WINEDEBUG=-all
+
+    if [ ! -d "$WINEPREFIX" ]; then
+      mkdir -p "$WINEPREFIX"
+      ${lib.getExe' wine "wineboot"} -u
+    fi
+
+    exec "$WINE" "''${ENTRYPOINT:-${lib.getExe dnspy}}" "$@"
+  '';
+
+  self = stdenvNoCC.mkDerivation {
+    inherit (dnspy) pname version;
+
+    dontUnpack = true;
+    dontConfigure = true;
+    dontBuild = true;
+
+    nativeBuildInputs = [
+      makeWrapper
+      copyDesktopItems
+    ];
+
+    installPhase = ''
+      runHook preInstall
+
+      icon_name="dnSpy${lib.optionalString is32Bit "-x86"}"
+      ${lib.getExe' buildPackages.icoutils "icotool"} -x "${dnspy.src}/dnSpy/dnSpy/Images/$icon_name.ico"
+      for f in "$icon_name"_*.png; do
+        res=$(basename "$f" | cut -d "_" -f3 | cut -d "x" -f1-2)
+        install -vD "$f" "$out/share/icons/hicolor/$res/apps/$pname.png"
+      done
+
+      install -vD "${wrapper}" "$out/bin/${self.meta.mainProgram}"
+
+      runHook postInstall
+    '';
+
+    desktopItems = makeDesktopItem {
+      name = self.pname;
+      desktopName = "dnSpy" + (lib.optionalString is32Bit " (32-bit)");
+      comment = self.meta.description;
+      icon = dnspy.pname;
+      exec = self.meta.mainProgram;
+      categories = [ "Development" ];
+    };
+
+    passthru = {
+      unwrapped = dnspy;
+
+      updateScript = writeShellScript "update-dnspy" ''
+        ${lib.getExe nix-update} "dnspy.unwrapped"
+        "$(nix-build -A "$UPDATE_NIX_ATTR_PATH.unwrapped.fetch-deps" --no-out-link)"
+      '';
```

Comment:
Can't this be in a separate shell file with the `nix-shell` shebang instead of `writeShellScript`?
It'd enable analysis with shell check.

Reply:
I don't think it's worth it for 2 lines. I think the real solution is making nix-update run fetch-deps? Other packages would benefit from it aswell

Reply:
True, given that we now have `nix-update` it'd be good to make it automagically run `fetch-deps`.
I'll see if I can't get a PR running for it eventually.

Reply:
take a look at `tone.updateScript`:

```
#!/usr/bin/env nix-shell
#!nix-shell --pure -i bash -p bash nix nix-update git cacert
set -eo pipefail

prev_version=$(nix eval --raw -f. tone.version)
nix-update tone
[[ $(nix eval --raw -f. tone.version) == "$prev_version" ]] ||
  "$(nix-build . -A tone.fetch-deps --no-out-link)"
````

I think it's important not to run fetch-deps unconditionally.